### ISSUE TITLE
Does "deferred field" imply under-the-hood `.defer()`?

### ISSUE THREAD

**Issue by danlynkew** [rsinger86/drf-flex-fields#136]
Django provides [the `.defer()` optimization for querysets](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#defer). This optimization drops the deferred field from the SQL query sent to the database. That is, `Country.objects.first().defer('states')` means "don't even ask the database for the country's `states` field."

**drf-flex-fields**'s documentation reads:
> Alternatively, you could treat `country` as a "deferred" field by not defining it among the default fields. To make a field deferred, only define it within the serializer's `expandable_fields`.

When **drf-flex-fields** is used to "defer" a field in this way, is Django's `.defer()` actually used under the hood?

Asked another way: will the default response for the below serializer retrieve a country's `states` field _from the database_ or not?
```
class CountrySerializer(FlexFieldsModelSerializer):
    class Meta:
        model = Country
        fields = ['name', 'population']

        expandable_fields = {
            'states': (StateSerializer, {'many': True})
        }
```

**Comment by danlynkew** [comment]
@rsinger86

**Comment by Caiofcas** [comment]
No, it does not use django's `.defer()`, it only means that it will not return that field in the serializer answer. You can confirm that by searching the source code : https://github.com/search?q=repo%3Arsinger86%2Fdrf-flex-fields%20defer&type=code

I know this question is a bit old but leaving it here for posterity.
